src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -15,39 +15,153 @@
 #include "deps_format.h"
 
 
+void get_all_fx_versions(
+    host_mode_t mode,
+    const pal::string_t& own_dir,
+    const pal::string_t& fx_name,
+    std::vector<pal::string_t>* fx_versions)
+{
+    // No FX resolution for standalone apps
+    if (mode == host_mode_t::standalone)
+    {
+        trace::verbose(_X("Standalone mode detected. Not gathering shared FX locations"));
+        return;
+    }
+
+    // No FX resolution for mixed apps
+    if (mode == host_mode_t::split_fx)
+    {
+        trace::verbose(_X("Split/FX mode detected. Not gathering shared FX locations"));
+        return;
+    }
+
+    pal::string_t local_dir;
+    std::vector<pal::string_t> global_dirs;
+
+    // own_dir contains DIR_SEPARATOR appended that we need to remove.
+    pal::string_t own_dir_temp = own_dir;
+    remove_trailing_dir_seperator(&own_dir_temp);
+
+    std::vector<pal::string_t> hive_dir;
+    hive_dir.push_back(own_dir_temp);
+
+    if (pal::get_global_dotnet_dirs(&global_dirs))
+    {
+        for (pal::string_t dir : global_dirs)
+        {
+            if (dir != own_dir_temp)
+            {
+                hive_dir.push_back(dir);
+            }
+        }
+    }
+
+    for (pal::string_t dir : hive_dir)
+    {
+        auto fx_dir = dir;
+        trace::verbose(_X("Gathering FX locations in [%s]"), fx_dir.c_str());
+
+        append_path(&fx_dir, _X("shared"));
+        append_path(&fx_dir, fx_name.c_str());
+
+        if (pal::directory_exists(fx_dir))
+        {
+            std::vector<pal::string_t> versions;
+            pal::readdir(fx_dir, &versions);
+            for (const auto& ver : versions)
+            {
+                // Make sure we filter out any non-version folders.
+                fx_ver_t parsed(-1, -1, -1);
+                if (fx_ver_t::parse(ver, &parsed, false))
+                {
+                    trace::verbose(_X("Found FX version [%s]"), ver.c_str());
+                    fx_versions->push_back(ver);
+                }
+            }
+        }
+    }
+}
+
+void get_all_sdk_versions(
+    const pal::string_t& own_dir,
+    std::vector<pal::string_t>* sdk_versions)
+{
+    trace::verbose(_X("Gathering SDK locations in [%s]"), own_dir.c_str());
+
+    auto sdk_dir = own_dir;
+    append_path(&sdk_dir, _X("sdk"));
```

Comment:
Do we also need to check the `get_global_dotnet_dirs` here?  We do when resolving the sdk path in `fx_muxer_t::resolve_sdk_dotnet_path`

Reply:
Thanks. Done